test/integration/consul-container/test/observability/access_logs_test.go

```diff
@@ -0,0 +1,180 @@
+		t.Skip()
+	}
+
+	cluster := createCluster(t)
+
+	// Turn on access logs. Do this before starting the sidecars so that they inherit the configuration
+	// for their admin interface
+	proxyDefault := &api.ProxyConfigEntry{
+		Kind: api.ProxyDefaults,
+		Name: api.ProxyConfigGlobal,
+		AccessLogs: &api.AccessLogsConfig{
+			Enabled:    true,
+			JSONFormat: "{\"banana_path\":\"%REQ(X-ENVOY-ORIGINAL-PATH?:PATH)%\"}",
+		},
+	}
+	set, _, err := cluster.Agents[0].GetClient().ConfigEntries().Set(proxyDefault, nil)
+	require.NoError(t, err)
+	require.True(t, set)
+
+	serverService, clientService := createServices(t, cluster)
+	_, port := clientService.GetAddr()
+
+	// Validate Custom JSON
+	require.Eventually(t, func() bool {
+		libassert.HTTPServiceEchoes(t, "localhost", port, "banana")
+		client := libassert.ServiceLogContains(t, clientService, "\"banana_path\":\"/banana\"")
+		server := libassert.ServiceLogContains(t, serverService, "\"banana_path\":\"/banana\"")
+		return client && server
+	}, 20*time.Second, 1*time.Second)
+
+	// Validate Logs on the Admin Interface
+	serverSidecar, ok := serverService.(*libservice.ConnectContainer)
+	require.True(t, ok)
+	ip, port := serverSidecar.GetAdminAddr()
+
+	httpClient := cleanhttp.DefaultClient()
+	url := fmt.Sprintf("http://%s:%d/clusters?fruit=bananas", ip, port)
+	_, err = httpClient.Get(url)
+	require.NoError(t, err, "error making call to Envoy admin interface")
+
+	require.Eventually(t, func() bool {
+		return libassert.ServiceLogContains(t, serverService, "\"banana_path\":\"/clusters?fruit=bananas\"")
+	}, 15*time.Second, 1*time.Second)
+
+	// TODO: add a test to check that connections without a matching filter chain are logged
+
+	// Validate Listener Logs
+	proxyDefault = &api.ProxyConfigEntry{
+		Kind: api.ProxyDefaults,
+		Name: api.ProxyConfigGlobal,
+		AccessLogs: &api.AccessLogsConfig{
+			Enabled:             true,
+			DisableListenerLogs: true,
+			TextFormat:          "Orange you glad I didn't say banana: %REQ(X-ENVOY-ORIGINAL-PATH?:PATH)%, %RESPONSE_FLAGS%",
+		},
+	}
+
+	set, _, err = cluster.Agents[0].GetClient().ConfigEntries().Set(proxyDefault, nil)
+	require.NoError(t, err)
+	require.True(t, set)
+	time.Sleep(5 * time.Second) // time for xDS to propagate
+
+	// Validate Custom Text
+	_, port = clientService.GetAddr()
+	require.Eventually(t, func() bool {
+		libassert.HTTPServiceEchoes(t, "localhost", port, "orange")
+		client := libassert.ServiceLogContains(t, clientService, "Orange you glad I didn't say banana: /orange, -")
+		server := libassert.ServiceLogContains(t, serverService, "Orange you glad I didn't say banana: /orange, -")
+		return client && server
+	}, 60*time.Second, 500*time.Millisecond) // For some reason it takes a long time for the server sidecar to update
+
+	// TODO: add a test to check that connections without a matching filter chain are NOT logged
+
+}
+
+func createCluster(t *testing.T) *libcluster.Cluster {
```

Comment:
I think we can replace this method with a call to https://github.com/hashicorp/consul/blob/a1e2a4f8d6bef92aaddcabd42d046b258dbffb30/test/integration/consul-container/libs/topology/peering_topology.go#L152

it works for me in another [PR](https://github.com/hashicorp/consul-enterprise/pull/4091/files#diff-49f24a7c95c856d2b708203061008c49c3962dbe2f577d945508f72335465452R24)

Reply:
Nice! In the future I think we should remove peering from the name since it's good for everything.